app/controllers/records_controller.rb

```diff
@@ -62,17 +68,24 @@ class RecordsController < ApplicationController
 
   # delete
   get '/records/:id/delete' do
-    @record = Records.find(params[:id])
-    p "Record is retrieved: id = #{params[:id]}"
-    p "User id for requested record: #{current_user.id}"
-
-    check_access_to_records(@record, :'records/delete')
+    id = params[:id]
+    error_if_not_logged_in
+    @record = find_record(id)
+    erb :'records/delete'
   end
 
   delete '/records/:id/delete' do
-    redirect_if_not_logged_in
-    @record = Records.find(params[:id])
-    @record.delete
-    redirect to '/people_list'
+    id = params[:id]
+    error_if_not_logged_in
+    @record = find_record(id)
+    if @record.nil?
+      response.status = 404
+      erb :'errors/error_404'
+    else
+      @record.delete
+      response.status = 200
+      headers['Location'] = '/people_list'
+      erb :'success_templates/deleted_record'
+    end
```

Comment:
Not sure if such approach is applicable there. Please, leave comments about your suggestions.
Thank you in advance 😊

Reply:
For our application - it's ok